File: Rust/src/water_bottles.rs
pub fn num_water_bottles(mut num_bottles: i32, num_exchange: i32) -> i32 {
    let mut drank = 0;

    while num_bottles >= num_exchange {
        let rem = num_bottles % num_exchange;
        drank += num_bottles - rem;
        num_bottles = (num_bottles / num_exchange) + rem;
    }

    drank + num_bottles
}

File: JavaScript/src/water_bottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
function numWaterBottles(numBottles, numExchange) {
  let result = 0;

  while (numBottles >= numExchange) {
    const rest = numBottles % numExchange;
    result += numBottles - rest;
    numBottles = Math.floor(numBottles / numExchange) + rest;
  }

  result += numBottles;

  return result;
}

export { numWaterBottles };
